### ISSUE TITLE
Pack hardening: propose a reviewed pack diff from eval failures, gated on held-out scenarios

### ISSUE THREAD

**Issue by chaholl** [AltairaLabs/promptarena#159]
## Problem

When a pack fails its evals, the loop is manual: read the results, form a theory about why, edit the pack, re-run. The theory-forming step is the slow part, and it is the part a model is good at when handed structured failure data.

## Proposal

An opt-in loop:

```
run  →  analyse failures  →  propose a pack diff  →  re-run on held-out scenarios
                                                  →  report score delta  →  emit a PR
```

Two things have to be true for this to be worth having.

### 1. The output is a diff, not an applied change

The product of a run is a **pull request against the pack** — prompts and fragments, with the before/after eval delta in the body as the evidence. Reviewable, revertible, attributable, and it goes through whatever review the repo already has.

**Applying a change without review is out of scope at every version.** A tool that silently rewrites the thing under test is not a testing tool.

### 2. Held-out scenarios, enforced

Iterating a prompt until the tests pass, using those same tests as the objective, is training on the test set. It produces packs that look better and are not.

So: the optimiser sees a **train** split; the reported delta is measured on **held-out** scenarios it never saw. A candidate that improves train but not holdout is reported as overfitting and **rejected**, not offered.

This is the correctness condition, not an enhancement. Without it the feature is actively harmful.

## Scope

**v1 — prompt and fragment text only**

- [ ] Declare a train/holdout split in the arena project (default 70/30, seeded, and the seed recorded with the run)
- [ ] Run, and summarise the failures
- [ ] Propose a diff limited to `prompts` and `fragments`
- [ ] Re-run on holdout; report per-eval delta
- [ ] Emit a PR when holdout improves and no eval regresses beyond a stated tolerance; otherwise report why and stop
- [ ] `--dry-run` prints the diff without opening anything

Narrow on purpose: it is the highest-value surface, and the review gate matters more than the breadth of the edit.

**v2** — `skills`, `tools` bindings, `workflow` branches. Bigger blast radius; wants v1's gate proven first.

## Open

- **Non-determinism in the reported delta.** Two runs of the same holdout differ. The accept/reject gate needs either a stated tolerance or repeated runs — decide which before building the gate, because it changes the shape.
- Split declared in project YAML or as a run flag? YAML is reproducible; a flag is easier to adopt. Possibly both, with YAML winning.
- How much failure context the proposal step needs. Whole transcripts are expensive; assertion messages alone may be too thin.
- Git provider abstraction — start with one, but do not bake it into the loop.
